lib/testing/create-apollo-test-client.ts

```diff
@@ -0,0 +1,33 @@
+import { TestingModule } from '@nestjs/testing';
+import * as apolloServerTesting from 'apollo-server-testing';
+import { GraphQLModule, GraphQLFederationModule } from '..';
+import { ApolloServerBase } from 'apollo-server-core';
+
+/**
+ * Provides a single hook to run operations through the request pipeline,
+ * enabling the most thorough tests possible without starting up an HTTP server.
+ * @param testingModule the Nest.js testing module
+ * @returns a query and mutate function that can be used to run operations against the
+ * server instance. Currently, queries and mutations are the only operation
+ * types supported by createTestClient.
+ * @see https://www.apollographql.com/docs/apollo-server/testing/testing/
+ */
+export function createTestClient(
+  testingModule: TestingModule,
+): apolloServerTesting.ApolloServerTestClient {
+  return apolloServerTesting.createTestClient(getApolloServer(testingModule));
+}
+
+function getApolloServer(testingModule: TestingModule): ApolloServerBase {
```

Comment:
Instead of using the `TestingModule`, we could make it somewhat more generic and use the `INestApplicationContext` here instead

Reply:
Done.